docs/en/options.rst
============
Options
============

0bin's behavior can be adjusted with options passed using a configuration
file or directly using the command line. Some parameters are only available
in the configuration file.

If an option is not passed, 0bin will use the default value from the file
zerobin/default_settings.py.


Command line
==================

--host and --port
-------------------

The host and port on which to listen for incomming request. Usually 127.0.0.1
and 8000 to listen locally or 0.0.0.0 and 80 to listen from the outside.

Default: 127.0.0.1 and 8000

Setting file : HOST and PORT


--debug
----------

Display a listing of useful debugging information when something goes wrong
instead of showing the 500 error page.

In debug mode, the server also reload automatically any modified Python file;

Default: False

Configuration file equivalent: DEBUG

.. _user-and-group-en:

--user and --group
-------------------

The user and group the server will adopt after start up.

Useful when you run the command with admin rights to be able to listen to the
port 80, but you wish that the process do not have access to protected files.

--group is set to --user if not passed.

Default: None

Configuration file equivalent: USER and GROUP

--settings-file
-----------------

Path to the configuration file, if you use any.

Default: None

Configuration file equivalent: None


--compressed-static
--------------------

Serve minified static files (css and js). Use it in production to get a faster
web site.

Default: False

Configuration file equivalent: COMPRESSED_STATIC_FILES

--version and --help
--------------------

Display the help or the version of 0bin.

Default: None

Configuration file equivalent: None

Examples
----------

Production::

  sudo zerobin --host 0.0.0.0 --port 80 --user foo --compressed-static

Developpement::

  zerobin --debug --serve-static


Configuration file
====================

The configuration file should be an ordinary Python file, usually named
settings.py. It's used this way::

  zerobin --settings-file '/path/to/settings.py'

Any options passed to the command line will have priority on the ones in
the configuration file. The zerobin/default_settings.py can be used as an
example to create your own file. It's heavily commented.


DEBUG
-----

Display a listing of useful debugging information when something goes wrong
instead of showing the 500 error page.

In debug mode, the server also reload automatically any modified Python file;


Default: False

Command line equivalent: --debug

.. _static-root-en:

STATIC_FILES_ROOT
------------------

Asbolute path to the directory where 0bin is going to look for static files
(css, js and images).

Default:  "static" directory in the "zerobin" directory

Command line equivalent: None

COMPRESSED_STATIC_FILES
-------------------------

Serve minified static files (css and js). Use it in production to get a faster
web site.

Default: False

Command line equivalent: --compressed-static

PASTE_FILES_ROOT
-----------------

Absolute path to the directory in which 0bin is going to look save pastes.

Default: "static/content" direcotry in the "zerobin" directory

Command line equivalent: None

.. _template-dirs-en:

TEMPLATE_DIRS
--------------

List of absolute path to directories containing templates that 0bin uses to
generate the web site pages. The first list items have priotity on the later.

If you wish to use your own templates, add the directory containing them
at the beginning of the list::

  from zerobin.defauls_settings import TEMPLATE_DIRS

  TEMPLATE_DIRS = (
      '/directy/path/to/your/templates',
  ) + TEMPLATE_DIRS

Default:  "view" directory in the "zerobin" directory

Command line equivalent: None

HOST and PORT
-------------------


The host and port on which to listen for incomming request. Usually 127.0.0.1
and 8000 to listen locally or 0.0.0.0 and 80 to listen from the outside.

Default: 127.0.0.1 and 8000

Configuration file equivalent: --host and --port

USER and GROUP
-------------------

The user and group the server will adopt after start up.

Useful when you run the command with admin rights to be able to listen to the
port 80, but you wish that the process do not have access to protected files.

GROUP is set to USER if not passed.

Default: None

Configuration file equivalent: --user and --group

MENU
------

A list of 'name' + 'link' pairs used to buld the menu at the top of each page.

You can use a relative or absolute link, and even an email address.

Any email address will be automatically protected against spam.

Default::

  MENU = (
      ('Home', '/'),
      ('Download 0bin', 'https://github.com/sametmax/0bin'),
      ('Contact', 'mailto:your@email.com') # email
  )


Command line equivalent: None

MAX_SIZE
---------

Approximative value for a paste size limite.

Valeur approximative de limite de taille d'un paste.

Default = 500000 octets (500 ko)

Command line equivalent: None


docs/fr/options.rst
============
Options
============

Le comportement de 0bin peut être ajusté à l'aide d'options passées depuis un
fichier de configuration ou directement à la ligne de commande. Certains
paramètres sont uniquement disponibles dans le fichier de configuration.

Si une option n'est pas passée, 0bin utilise la valeur par
défaut présente dans le fichier zerobin/default_settings.py.

Ligne de commande
==================

--host et --port
-------------------

L'hôte et le port sur lesquels écouter les requêtes entrantes. En général
127.0.0.1 et 8000 pour écouter localement, ou 0.0.0.0 et 80 pour
écouter les requêtes extérieures.

Défaut: 127.0.0.1 et 8000

Equivalent du fichier de configuration : HOST et PORT


--debug
----------

Afffiche un listing d'informations utiles pour déboguer quand quelque chose
se passe mal à la place d'une page d'erreur 500.

En mode debug, le serveur recharge aussi automatiquement tout fichier Python
modifié.

Défaut: False

Equivalent du fichier de configuration : DEBUG

.. _user-and-group-fr:

--user et --group
-------------------

L'utilisateur et le group que le serveur essayera d'adopter après le démarrage.

Utile quand on lance la commande avec les droits admin afin de pouvoir
écouter sur le port 80, mais qu'on souhaite, pour des raisons de sécurité, que
le processus n'est pas accès aux fichiers protégés du système.

--group prend de --user si il n'est pas précisé.

Défaut: Aucun

Equivalent du fichier de configuration : USER et GROUP

--settings-file
-----------------

Chemin vers le fichier de configuration, si vous souhaitez en utiliser un.

Défaut: Aucun

Equivalent du fichier de configuration : Aucun


--compressed-static
--------------------

Sert les versions minifiées fichiers statiques (css et js). A utiliser en
production pour un site plus rapide.

Défaut: False

Equivalent du fichier de configuration : COMPRESSED_STATIC_FILES

--version et --help
--------------------

Affiche l'aide ou la version de 0bin

Défaut: Aucun

Equivalent du fichier de configuration : Aucun

Exemples
----------

Production::

  sudo zerobin --host 0.0.0.0 --port 80 --user foo --compressed-static

Développement::

  zerobin --debug --serve-static

Fichier de configuration
========================

Le fichier configuration doit être un fichier Python ordinnaire, généralement
appelé settings.py. On l'utilise ainsi::

  zerobin --settings-file '/chemin/vers/settings.py'

Toutes les autres options passées à la commandes auront priorité sur les options
du fichier de configuration. Le fichier zerobin/default_settings.py peut servir
d'exemple pour créer son propre fichier de configuration, il est largement commenté.

DEBUG
-----

Afffiche un listing d'informations utiles pour déboguer quand quelque chose
se passe mal à la place d'une page d'erreur 500.

En mode debug, le serveur recharge aussi automatiquement tout fichier Python
modifié.

Défaut: False

Equivalent en ligne de commande : --debug

.. _static-root-fr:

STATIC_FILES_ROOT
------------------

Chemin absolu du dossier dans lequel 0bin va chercher les fichiers statiques
(css, js et images).

Défaut: dossier "static" dans le dossier "zerobin"

Equivalent en ligne de commande : Aucun

COMPRESSED_STATIC_FILES
-------------------------

Sert les versions minifiées fichiers statiques (css et js). A utiliser en
production pour un site plus rapide.

Défaut: False

Equivalent en ligne de commande : --compressed-static

PASTE_FILES_ROOT
-----------------

Chemin absolu du dossier dans lequel 0bin va sauvegarder les pastes.

Défaut: dossier "static/content" dans le dossier "zerobin"

Equivalent en ligne de commande : Aucun

.. _template-dirs-fr:

TEMPLATE_DIRS
--------------

Liste des chemins absolus des dossiers qui contiennent les templates que 0bin
utilise pour générer les pages du site. Les premiers éléments de la liste
ont priorités sur les suivant.

Si vous voulez utiliser vos propres templates, ajoutez le dossier qui les
contient au début de la liste::

  from zerobin.defauls_settings import TEMPLATE_DIRS

  TEMPLATE_DIRS = (
      '/chemin/version/votre/dossier/de/templates',
  ) + TEMPLATE_DIRS

Défaut: dossier "view" dans le dossier "zerobin"

Equivalent en ligne de commande : Aucun


HOST et PORT
-------------------

L'hôte et le port sur lesquels écouter les requêtes entrantes. En général
127.0.0.1 et 8000 pour écouter localement, ou 0.0.0.0 et 80 pour
écouter les requêtes extérieures.

Défaut: 127.0.0.1 et 8000

Equivalent du fichier de configuration : --host et --port

USER et GROUP
-------------------

L'utilisateur et le group que le serveur essayera d'adopter après le démarrage.

Utile quand on lance la commande avec les droits admin afin de pouvoir
écouter sur le port 80, mais qu'on souhaite, pour des raisons de sécurité, que
le processus n'est pas accès au fichiers système.

GROUP prend de USER si il n'est pas précisé.

Défaut: Aucun

Equivalent en ligne de commande : --user et --group

MENU
------

Une liste de paires 'Nom' + 'Liens' à utiliser pour construire le menu qui
est en haut de chaque page.

Le lien peut être un lien relatif, absolu, ou une adresse email.

Toute addresse email sera automatiquement protégée contre le spam.

Défaut::

  MENU = (
      ('Home', '/'),
      ('Download 0bin', 'https://github.com/sametmax/0bin'),
      ('Contact', 'mailto:your@email.com') # email
  )

Equivalent en ligne de commande : Aucun

MAX_SIZE
---------

Valeur approximative de limite de taille d'un paste.

Défaut = 500000 octets (500 ko)

Equivalent en ligne de commande : Aucun

settings.py
# -*- coding: utf-8 -*-

# import default settings value from src/default_settings.py
# you can refer to this file if you forgot what
# settings is for and what it is set to by default
# DO NOT ALTER THIS LINE
from src.default_settings import *

# debug will get you error message and auto reload
# don't set this to True in production
DEBUG = False

# absolute path where the paste files should be store
# default in projectdirectory/static/content/
# use "/" even under Windows
PASTE_FILES_ROOT = os.path.join(STATIC_FILES_ROOT, 'content')

# Port and host the embeded python server should be using
# You can also specify them using the --host and --port script options
# which have priority on these settings
HOST = "127.0.0.1"
PORT = "8000"

# User and group the server should run as. Set to None if it should be the
# current user. Some OS don't support it and if so, it will be ignored.
USER = None
GROUP = None

# limit size of pasted text in bytes. Be carefull allowing too much
# size can slow down user's browser
MAX_SIZE = 1024 * 500
MAX_SIZE_KB = int(math.ceil(MAX_SIZE / 1024.0))

# Names/links to insert in the menu bar.
# Any link with "mailto:" will be escaped to prevent spam
MENU = (
    ('Home', '/'), # internal link
    ('Download 0bin', 'https://github.com/sametmax/0bin'), # external link
    ('Contact', 'mailto:your@email.com') # email
)

# this import a file named settings_local.py if it exists
# you may want to create such a file to have different settings
# on each machine
try:
    from settings_local import *
except ImportError:
    pass
